frontend/src/component/project/Project/hooks/useProjectForm.test.ts

```diff
@@ -93,13 +113,134 @@ describe('payload generation', () => {
 });
 
 describe('name validation', () => {
+    const existingProjects = [
+        { id: 'project1', name: 'Project One' },
+        { id: 'project2', name: 'Project Two' },
+    ];
+
     test.each([
         ['An empty string', ''],
         ['Just whitespace', '     '],
     ])(`%s is not valid`, (_, value) => {
         const { result } = renderHook(() => useProjectForm());
 
-        result.current.setProjectName(value);
-        expect(result.current.validateName()).toBeFalsy();
+        act(() => {
+            result.current.setProjectName(value);
+        });
+
+        let isValid = true;
+        act(() => {
+            isValid = result.current.validateName();
+        });
+
+        expect(isValid).toBeFalsy();
+    });
+
+    test('accepts a unique project name on creation', () => {
+        mockUseProjects.mockReturnValue(
+            createProjectsResponse(existingProjects),
+        );
+
+        const { result } = renderHook(() => useProjectForm());
+
+        act(() => {
+            result.current.setProjectName('Brand New Project');
+        });
+
+        let isValid = false;
+        act(() => {
+            isValid = result.current.validateName();
+        });
+
+        expect(isValid).toBeTruthy();
+        expect(result.current.errors).not.toHaveProperty('name');
+    });
+
+    test('rejects duplicate project name on creation', () => {
+        mockUseProjects.mockReturnValue(
+            createProjectsResponse(existingProjects),
+        );
+
+        const { result } = renderHook(() => useProjectForm());
+
+        act(() => {
+            result.current.setProjectName('Project One');
+        });
+
+        let isValid = true;
+        act(() => {
+            isValid = result.current.validateName();
+        });
+
+        expect(isValid).toBeFalsy();
+        expect(result.current.errors).toHaveProperty(
+            'name',
+            'This name is already taken by a different project.',
+        );
+    });
+
+    test('allows keeping the original name when editing', () => {
+        mockUseProjects.mockReturnValue(
+            createProjectsResponse(existingProjects),
+        );
+
+        const { result } = renderHook(() =>
+            useProjectForm('project1', 'Project One'),
+        );
+
+        let isValid = false;
+        act(() => {
+            isValid = result.current.validateName();
+        });
+
+        expect(isValid).toBeTruthy();
+        expect(result.current.errors).not.toHaveProperty('name');
+    });
```

Comment:
I'm not sure what this test is doing, tbh. I think it's more important to check that you can keep the current name, even if it's already taken by a different project. E.g. something like:

```suggestion
    test('allows keeping the original name when editing', () => {
        mockUseProjects.mockReturnValue(
            createProjectsResponse(existingProjects),
        );

        const { result } = renderHook(() =>
            useProjectForm('project3', 'Project One'),
        );

        expect(result.current.validateName()).toBeTruthy();
    });
```